---
title: "Scholarly Publications"
---

# Pre-Prints

> These publications have not yet been peer-reviewed, but have been submitted.

**Kamvar ZN**, Cai J, Pulliam JRC, Schumacher J, Jombart T. Epidemic curves made easy using the R package *incidence* [version 1; referees: awaiting peer review]. F1000Research 2019, 8:139. (doi: [10.12688/f1000research.18002.1](https://doi.org/10.12688/f1000research.18002.1)).

 - software: https://github.com/reconhub/aweek
 - doi:      [10.5281/zenodo.1404718](https://doi.org/10.5281/zenodo.1404718)

# Peer-Reviewed Publications

Moraga P, Dorigatti I, **Kamvar ZN**, Piatkowski P, Toikkanen SE, Nagraj V, Donnelly CA, and Jombart T epiflows: an R package for risk assessment of travel-related spread of disease \[version 1;  referees: 2 approved with reservations\]. F1000Research 2018, 7:1374 (doi: [10.12688/f1000research.16032.1](https://doi.org/10.12688/f1000research.16032.1))

 -  software:  https://github.com/reconhub/epiflows
 -  doi:       [10.5281/zenodo.1401806](http://doi.org/10.5281/zenodo.1401806)

Miorini TJJ, **Kamvar ZN**, Higgins R, Raetano CG, Steadman JR, Everhart SE. (2018) Variation in pathogen aggression and cultivar performance against *Sclerotinia sclerotiorum* in soybean and dry bean from Brazil and the U.S. *Tropical Plant Pathology* doi: [10.1007/s40858-018-00273-w](https://doi.org/10.1007/s40858-018-00273-w)

 -  data/analysis:  https://github.com/everhartlab/SscPhenoProj                    
 -  doi:            [10.17605/OSF.IO/2X7FC](https://doi.org/10.17605/OSF.IO/2X7FC) 

Pannullo A, **Kamvar ZN**, Miorini TJJ, Steadman JR, Everhart SE. (2018) Genetic variation and structure of *Sclerotinia sclerotiorum* populations from soybean in Brazil. *Tropical Plant Pathology* doi: [10.1007/s40858-018-0266-5](https://doi.org/10.1007/s40858-018-0266-5).

 -  data/analysis:  https://github.com/everhartlab/brazil-sclerotinia-2017         
 -  doi:            [10.17605/OSF.IO/E4UPH](https://doi.org/10.17605/OSF.IO/E4UPH) 

**Kamvar ZN**, Everhart SE. (2018) Something in the agar does not compute: on the discriminatory power of mycelial compatibility in *Sclerotinia sclerotiorum*. *Tropical Plant Pathology*  doi: [10.1007/s40858-018-0263-8](https://doi.org/10.1007/s40858-018-0263-8)    
[Download the un-formatted PDF](https://github.com/everhartlab/sclerotinia-review-2017/raw/master/manuscript/review.pdf)

 -  simulations:  https://github.com/everhartlab/sclerotinia-review-2017           
 -  doi:          [10.17605/OSF.IO/B8VD3](https://doi.org/10.17605/OSF.IO/B8VD3) 

**Kamvar ZN**, Amaradasa BS, Jhala R, McCoy S, Steadman JR, Everhart SE. (2017) Population structure and phenotypic variation of *Sclerotinia sclerotiorum* from dry bean (*Phaseolus vulgaris*) in the United States. *PeerJ* 5:e4152 doi: [10.7717/peerj.4152](https://doi.org/10.7717/peerj.4152)

 -  data/analysis:  https://github.com/everhartlab/sclerotinia-366                 
 -  doi:            [10.17605/OSF.IO/EJB5Y](https://doi.org/10.17605/OSF.IO/EJB5Y) 

Grünwald NJ, Everhart SE, Knaus BJ, **Kamvar ZN**. (2017) Best practices for population genetic analyses. *Phytopathology* 107:9 doi: [10.1094/PHYTO-12-16-0425-RVW](https://doi.org/10.1094/PHYTO-12-16-0425-RVW)

Jombart T, Archer F, Schliep K, **Kamvar ZN**, Harris R, Paradis E, Goudet J, Lapp H (2017). apex: phylogenetics with multiple genes. *Molecular Ecology Resources.* 17:1 19-26 doi: [10.1111/1755-0998.12567](https://doi.org/10.1111/1755-0998.12567)

**Kamvar ZN**, López-Uribe MM, Coughlan S, Grünwald NJ, Lapp H, Manel S (2017). Developing educational resources for population genetics in R: an open and collaborative approach. *Molecular Ecology Resources.* 17:1 120-128 doi: [10.1111/1755-0998.12558](https://doi.org/10.1111/1755-0998.12558)

Tabima JF, Everhart SE, Larsen MM, Weisberg AJ, **Kamvar ZN**, Tancos MA, Smart CD, Chang JH, Grünwald NJ. (2016) Microbe-ID: an open source toolbox for microbial genotyping and species identification. *PeerJ* 4: e2279 doi: [10.7717/peerj.2279](https://doi.org/10.7717/peerj.2279)

Grünwald NJ, Larsen MM, **Kamvar ZN**, Reeser PW, Kanaskie A, Laine J, Wiese R (2015) First report of the EU1 clonal lineage of *Phytophthora ramorum* on tanoak in an OR forest. *Plant Disease.* 100:5, 1024-1024. doi: [10.1094/PDIS-10-15-1169-PDN](https://doi.org/10.1094/PDIS-10-15-1169-PDN)

**Kamvar ZN**, Brooks JC and Grünwald NJ (2015) Novel R tools for analysis of genome-wide population genetic data with emphasis on clonality. *Front. Genet.* 6: 208. doi: [10.3389/fgene.2015.00208](https://doi.org/10.3389/fgene.2015.00208)

 -  data/analysis:  https://github.com/grunwaldlab/supplementary-poppr-2.0        
 -  doi:            [10.5281/zenodo.17424 ](https://doi.org/10.5281/zenodo.17424) 

**Kamvar ZN**, Larsen MM, Kanaskie AM, Hansen EM, and Grünwald NJ. (2015) Spatial and temporal analysis of populations of the sudden oak death pathogen in Oregon forests. *Phytopathology.* 105:7 982-989. doi: [10.1094/PHYTO-12-14-0350-FI.](https://doi.org/10.1094/PHYTO-12-14-0350-FI.)

 -  data/analysis:  https://github.com/zkamvar/Sudden_Oak_Death_in_Oregon_Forests 
 -  doi:            [10.5281/zenodo.13007 ](https://doi.org/10.5281/zenodo.13007) 

Weiland JE, Garrido PA, **Kamvar ZN**, Marek SM, Grünwald NJ, and Garzón CD. (2015) Population structure of *Pythium irregulare*, *P. sylvaticum*, and *P. ultimum* in forest nursery soils of Oregon and Washington. *Phytopathology.* 105:5 684-694. doi: [10.1094/PHYTO-05-14-0147-R](https://doi.org/10.1094/PHYTO-05-14-0147-R)

**Kamvar ZN**, Tabima JF, Grünwald NJ. (2014) *Poppr*: an R package for genetic analysis of populations with clonal, partially clonal, and/or sexual reproduction. *PeerJ* 2: e281. doi: [10.7717/peerj.281](https://doi.org/10.7717/peerj.281)

Buckner B, Beck J, Browning, K, Hoxha E, Grantham L, **Kamvar ZN**, Lough A, Nikolova O, and Schnable PS, Scanlon MJ, and Janick-Buckner D. (2007) Involving undergraduates in the annotation and analysis of global gene expression studies: creation of a maize shoot apical meristem expression database. *Genetics* 176: 741-747. doi: [10.1534/genetics.106.066472](https://doi.org/10.1534/genetics.106.066472)
